favorite_files.py

```diff
@@ -112,13 +112,17 @@ def add(self, names, group_name=None):
         added = 0
         # Iterate names and add them to group/global if not already added
         for n in names:
-            if not Favs.exists(n, group_name=group_name):
+            if Favs.file_index(n, group_name=group_name) is None:
                 if exists(n):
                     Favs.set(n, group_name=group_name)
                     added += 1
                 else:
                     # File does not exist on disk; cannot add
                     disk_omit_count += 1
+            else:
+                message = "is already among your favorite files and has not been added."
+                print(n, message)   # full filename printed to the console
+                sublime.active_window().status_message("Some file " + message)
```

Comment:
Why `Some file`?  Did you mean to use the base filename here?  Also, as mentioned above, please use `error`.

Reply:
I forgot to explain this.. I wanted to avoid using error() because it triggers an error toolbox. More files can be already present and I wanted to avoid toolbox spamming. This only notifies in the status bar that some files haven't been added, and prints the details for the individual files in the console. The log() function would be better, I agree.

Anyway I added this primarily to test that everything was ok, I'm not meaning it should stay in the final version. If you find it appropriate it can stay, otherwise it can go. It's still some feedback on something has happened, I don't think it hurts, but using error() definitely would bother me, it can easily happen that there are duplicates if you add all files in the window. If you don't like it I remove it though, or I use log().

Reply:
Cool, that's fine for now.